olive/engine/engine.py

```diff
@@ -90,6 +93,38 @@ def __init__(
         else:
             self.target = LocalSystem()
 
+        if execution_providers is None:
+            execution_providers = self._config.execution_providers
+
+        # verfiy the AzureML system have specified the execution providers
+        # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
+        if self.target.system_type == SystemType.AzureML and execution_providers is None:
+            raise ValueError("AzureMLSystem requires execution providers to be specified.")
+        elif execution_providers is None:
+            if self.target.system_type in (SystemType.Local, SystemType.PythonEnvironment):
+                execution_providers = self.target.get_supported_execution_providers()
+            else:
+                # for docker system and python system, we default use CPUExecutionProvider
+                execution_providers = ["CPUExecutionProvider"]
+
+        self.execution_providers = execution_providers
+
+        # Flatten the accelerators to list of AcceleratorSpec
+        accelerators: List[str] = self.target.accelerators
+        if accelerators is None:
+            logger.warning("No accelerators specified for target system. Using CPU.")
+            accelerators = ["CPU"]
+
+        self.accelerator_specs: List[AcceleratorSpec] = []
+        for accelerator in accelerators:
+            device = Device(accelerator.lower())
+            supported_eps = AcceleratorLookup.get_execution_providers_for_device(device)
```

Comment:
Nitpick: Looks like device and accelerator is the same concept, right? How about align the name to accelerator? Like:
`get_execution_providers_for_accelerator`.